### PR TITLE
Use Pillow for default submission thumbnails

### PR DESCRIPTION
**Pull request by charmander**
Halves JPEG thumbnail sizes; fixes longstanding colour profile (?) bug; improves thumbnail quality. Also allows more extreme aspect ratios on images that are smaller than thumbnails.

If this works well, it can be applied to characters, covers, avatars, custom thumbnails, and past thumbnails, saving some amount of disk space and bandwidth.

### CODE REVIEW

**Comment by charmander** [comment]
@hyena This part of the implementation uses only the resized first frame of an animation as its thumbnail, matching the existing sanpera behaviour. Extending PIL to animated covers is more involved (https://pillow.readthedocs.io/en/latest/reference/ImageSequence.html, https://pillow.readthedocs.io/en/5.1.x/handbook/image-file-formats.html#gif), but supported.

File: libweasyl/libweasyl/images_new.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# encoding: utf-8
+
+"""
+Image manipulation with Pillow.
+"""
```
Comment (hyena) [inline libweasyl/libweasyl/images_new.py]: Add a todo here to rename this file when the switchover is complete.

File: weasyl/submission.py [hunk 2]
```diff
@@ -217,8 +230,11 @@ def create_visual(userid, submission,
         submitid, 'submission', submit_media_item)
     orm.SubmissionMediaLink.make_or_replace_link(
         submitid, 'cover', cover_media_item)
-    orm.SubmissionMediaLink.make_or_replace_link(
-        submitid, 'thumbnail-generated', thumb_generated_media_item)
+
+    for thumb_generated_media_item in thumb_generated_media_items:
+        orm.SubmissionMediaLink.make_or_replace_link(
```
Comment (charmander) [inline weasyl/submission.py]: If `thumb_generated_media_items` has multiple elements, is this just going to replace the same link repeatedly?

**Comment by charmander** [comment]
Okay! Let’s just do it.

**Pull request merged**